src/Controller/CheckboxController.php

```diff
@@ -5,24 +5,41 @@
 namespace App\Controller;
 
 use App\Entity\Checkbox;
-use App\Entity\Project;
 use App\Entity\ProjectCheckbox;
-use Doctrine\ORM\EntityManagerInterface;
+use App\Repository\CheckListRepository;
+use App\Repository\ProjectCheckListRepository;
+use App\Repository\ProjectRepository;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\JsonResponse;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Routing\Annotation\Route;
 
 final class CheckboxController extends AbstractController
 {
     /**
-     * @var EntityManagerInterface
```

Comment:
There should be clear contract and layer separation:

- Controllers should only use Repositories
- Repositories should only use EntityManager

Maybe PHPStan would help use here